packages/affiliates/libs/affiliates/devmining.js

```diff
@@ -12,16 +12,18 @@ const { EmpBalancesHistory } = require("../processors");
 const { Prices } = require("../models");
 const { DecodeLog } = require("../contracts");
 
-module.exports = ({ queries, empAbi, coingecko, synthPrices, firstEmpDate }) => {
+module.exports = ({ queries, defaultEmpAbi, coingecko, synthPrices, firstEmpDate }) => {
   assert(queries, "requires queries class");
-  assert(empAbi, "requires empAbi");
+  assert(defaultEmpAbi, "requires defaultEmpAbi");
   assert(coingecko, "requires coingecko api");
   assert(synthPrices, "requires synthPrices api");
 
   // use firstEmpDate as a history cutoff when querying for events. We can safely say no emps were deployed before Jan of 2020.
   firstEmpDate = firstEmpDate || moment("2020-01-01", "YYYY-MM-DD").valueOf();
 
-  async function getBalanceHistory(empAddress, start, end) {
+  async function getBalanceHistory(empAddress, start, end, empAbi = defaultEmpAbi) {
+    assert(empAddress, "requires empAddress");
+    assert(empAbi, "requires empAbi");
```

Comment:
Is this check just to guard against null being passed in? Because empAbi has a default...?

Reply:
it was to maintain backward compatibilility if you did not provide an explicit emp Abi. But I think i might pass it in the way you recommended on first comment